pandas/_typing.py

```diff
@@ -70,6 +73,8 @@
 else:
     # typing.final does not exist until py38
     final = lambda x: x
+    # typing.TypedDict does not exist until py38
+    TypedDict = dict
```

Comment:
if we don't need to use the TypedDict at runtime, then we would not need this and we could only use TypedDict in TYPE_CHECKING blocks. 

I see the downside being that aliases (and imports of aliases) would also need to be defined in the TYPE_CHECKING block and that we would not be able to construct the TypedDict explicitly in code. but that could be an upside.

no strong preference here yet.

Reply:
thinking some more. leave this for now.

from NEP 29,  Dec 26, 2021 drop support for Python 3.7

so we should be able to drop python 3.7 on master after the 1.3 release.